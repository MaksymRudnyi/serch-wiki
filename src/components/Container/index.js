import React from 'react';
import {useDebounce, useSearch, useSearchForm} from "../../hooks";

const Container = ({ children }) => {
    const {searchValue, onSearchChange} = useSearchForm();
    const {articles} = useSearch(useDebounce(searchValue));

    return children({ searchValue, onSearchChange, articles })
};

export default Container;