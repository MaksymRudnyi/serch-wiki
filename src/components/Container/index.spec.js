import React from 'react';
import {mount} from 'enzyme'

import Container from ".";

jest.mock('../../hooks', () => ({
    useDebounce: jest.fn(() => ({})),
    useSearch: jest.fn(() => ({articles: []})),
    useSearchForm: jest.fn(() => ({
        searchValue: 'testValue',
        onSearchChange: jest.fn()
    }))
}));

const render = (props, children) => mount(<Container {...props}>{children}</Container>);

const Children = () => <div>Test</div>;

describe('Container component', () => {
    let sut;
    let props;
    let children = ({searchValue, onSearchChange, articles}) => <Children searchValue={searchValue}
                                                                          onSearchChange={onSearchChange}
                                                                          articles={articles}/>;

    describe('without props', () => {
        beforeEach(() => {
            sut = render(props, children)
        });

        it('should match snapshot', () => {
            expect(sut).toMatchSnapshot();
        })
    });

    describe('when passed custom props', () => {
        beforeEach(() => {
            sut = render(props, children)
        });

        it('should return searchValue in Children component', () => {
            const { searchValue } = sut.find('Children').props();

            expect(searchValue).toBe('testValue')
        })
    })
})