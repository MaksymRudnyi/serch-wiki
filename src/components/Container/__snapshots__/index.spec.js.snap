// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component without props should match snapshot 1`] = `
<Container>
  <Children
    articles={Array []}
    onSearchChange={[MockFunction]}
    searchValue="testValue"
  >
    <div>
      Test
    </div>
  </Children>
</Container>
`;
